### PR TITLE
[StarRocks On ES] [Bugfix] nested or object type field would degrade to source fetch

### LINKED ISSUE #6625: [StarRocks On ES] nested or object type field return null value with `varchar` type
### Steps to reproduce the behavior (Required)
1. Create ES Index

```
PUT nested-type-test
{
   "settings": {
      "index": {
         "number_of_replicas": 0,
         "number_of_shards": 1
      }
   },
   "mappings": {
      "properties": {
         "id": {
            "type": "long"
         },
         "k1": {
            "type": "nested"
         },
         "k2": {
            "properties": {
               "k21": {
                  "type": "keyword"
               },
               "k22": {
                  "type": "keyword"
               }
            }
         }
      }
   }
}
```

3. index some documents

```
POST nested-type-test/_doc
{
    "id": 1,
    "k1": {
        "k11": "v11",
        "k22": "v22"
    },
    "k2": {
        "k21": "v21",
        "k22": "v22"
    }
}
```
4. create external table 
```
CREATE EXTERNAL TABLE `test1` (
  `k1` varchar(20),
  `k2` varchar(20)
) ENGINE=ELASTICSEARCH
PROPERTIES (
"hosts" = "http://172.26.195.67:8201",
"index" = "nested-type-test",
"type" = "_doc"
);
```
5. query the test1

```
select * from test1;
```
### Expected behavior (Required)

MySQL [wyf]> select * from test1;
+---------------------------+---------------------------+
| k1                        | k2                        |
+---------------------------+---------------------------+
| {"k11":"v11","k22":"v22"} | {"k22":"v22","k21":"v21"} |
+---------------------------+---------------------------+
### Real behavior (Required)
+---------------------------+---------------------------+
| k1                        | k2                        |
+---------------------------+---------------------------+
| NULL                  | NULL                  |
+---------------------------+---------------------------+
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
all verion

### PR DESCRIPTION
**Pull request by RowenWoo**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes https://github.com/StarRocks/starrocks/issues/6625

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/elasticsearch/MappingPhase.java|1|2|50.00%|

File: fe/fe-core/src/main/java/com/starrocks/external/elasticsearch/MappingPhase.java [hunk 1]
```diff
@@ -130,6 +130,10 @@ private void resolveKeywordFields(SearchContext searchContext, JSONObject fieldO
 
     private void resolveDocValuesFields(SearchContext searchContext, JSONObject fieldObject, String colName) {
         String fieldType = fieldObject.optString("type");
+        // do not process `nested` or object type
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/external/elasticsearch/MappingPhase.java:133]:
```suggestion
        // Not process `nested` or object type
```

**Review by murphyatwork — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/external/elasticsearch/MappingPhase.java (hunk 1 above)
Comment (RowenWoo) [reply fe/fe-core/src/main/java/com/starrocks/external/elasticsearch/MappingPhase.java:133]: maybe use skip or ignore is better

**Review by imay — APPROVED** [review]

**Review by murphyatwork — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/elasticsearch/MappingPhase.java|1|2|50.00%|

**Comment by RowenWoo** [comment]
https://github.com/Mergifyio backport branch-2.2 branch-2.1 branch-2.0

**Comment by RowenWoo** [comment]
@Mergifyio backport branch-2.2 branch-2.1 branch-2.0

**Pull request merged**